src/dbnode/storage/block/block.go

```diff
@@ -96,6 +98,10 @@ func NewDatabaseBlock(
 	return b
 }
 
+func (b *dbBlock) SetNamespaceContext(nsCtx namespace.Context) {
```

Comment:
Can we make this part of the `New(..)` and the `Reset(...)` calls to?

Reply:
I can merge this to Reset, as the block is taken out of the pool. (similar to SetSchema into Reset).